Source/WebCore/PAL/pal/PALSwift/UnsafeOverlays.swift

```diff
@@ -30,12 +30,92 @@ import CryptoKit
 
 import PALSwift
 
+enum UnsafeErrors: Error {
+    case invalidLength
+    case emptySpan
+}
+
 extension HashFunction {
     mutating func update(data: SpanConstUInt8) {
-        self.update(
-            bufferPointer: UnsafeRawBufferPointer(
-                start: data.__dataUnsafe(), count: data.size()))
+        if data.empty() {
+            self.update(data: Zero.zeroData)
+        } else {
+            self.update(
+                bufferPointer: UnsafeRawBufferPointer(
+                    start: data.__dataUnsafe(), count: data.size()))
+        }
+    }
+}
+
+// Ugly but needed.
+// all __dataUnsafe() calls here return nil if input WTF::Vector is zero length.
+// CommonCrypto APIs handled nil pointers internally. Swift APIs crash.
+// To maintain backward compatibility, a zero array is needed to go past Swift Validation
+// and let CyptoKit replicate behavior.
```

Comment:
I would say, "CryptoKit does not support a null pointer with zero length. We instead need to pass an empty Data. This class provides that."